scripts/docker/zeppelin-server/Dockerfile

```diff
@@ -0,0 +1,93 @@
+    /sbin/apk add --no-cache wget gnupg
+    # Install tini
+RUN set -ex && \
+    /usr/bin/wget -q https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-static-amd64 -O /sbin/tini && \
+    /usr/bin/wget -q https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-static-amd64.asc -O /tmp/tini.asc && \
+    /usr/bin/gpg --batch --keyserver keyserver.ubuntu.com --recv-keys 595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7 && \
+    /usr/bin/gpg --batch --verify /tmp/tini.asc /sbin/tini && \
+    /bin/chmod +x /sbin/tini && \
+    /bin/rm -v /tmp/tini.asc
+
+FROM ubuntu:18.04
+LABEL maintainer="Apache Software Foundation <dev@zeppelin.apache.org>"
+
+RUN set -ex && \
+    apt-get -y update && \
+    # Install language and other base packages
+    DEBIAN_FRONTEND=noninteractive apt-get install -y language-pack-en openjdk-8-jre-headless wget apt-transport-https gnupg && \
+    apt-key adv --fetch-keys https://packages.cloud.google.com/apt/doc/apt-key.gpg && \
+    echo "deb https://apt.kubernetes.io/ kubernetes-xenial main" | tee -a /etc/apt/sources.list.d/kubernetes.list && \
+    DEBIAN_FRONTEND=noninteractive apt-get -y update && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y kubectl && \
+    # Cleanup
+    rm -rf /var/lib/apt/lists/* && \
+    apt-get autoclean && \
+    apt-get clean
+
+ARG version="0.9.0-SNAPSHOT"
+
+ENV LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8 \
+    JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 \
+    VERSION="${version}" \
+    Z_HOME="/opt/zeppelin" \
+    ZEPPELIN_ADDR="0.0.0.0" \
+    ZEPPELIN_WAR_TEMPDIR="/tmp/webapps" \
+    ZEPPELIN_DEP_LOCALREPO="/tmp/local-repo"
+
+# Copy Zeppelin related files
+COPY --from=zeppelin-distribution /opt/zeppelin/zeppelin-web-${VERSION}.war ${Z_HOME}/
+COPY --from=zeppelin-distribution /opt/zeppelin/zeppelin-web-angular-${VERSION}.war ${Z_HOME}/
+COPY --from=zeppelin-distribution /opt/zeppelin/conf ${Z_HOME}/conf
+COPY --from=zeppelin-distribution /opt/zeppelin/bin ${Z_HOME}/bin
+COPY --from=zeppelin-distribution /opt/zeppelin/lib ${Z_HOME}/lib
+COPY --from=zeppelin-distribution /opt/zeppelin/plugins ${Z_HOME}/plugins
+COPY --from=zeppelin-distribution /opt/zeppelin/interpreter/zeppelin-interpreter-shaded-${VERSION}.jar ${Z_HOME}/interpreter/zeppelin-interpreter-shaded-${VERSION}.jar
+# copy example notebooks
+COPY --from=zeppelin-distribution /opt/zeppelin/notebook ${Z_HOME}/notebook
+# copy k8s files
+COPY --from=zeppelin-distribution /opt/zeppelin/k8s ${Z_HOME}/k8s
+# Copy interpreter-settings for activate interpreter
+# COPY --from=zeppelin-distribution /opt/zeppelin/interpreter/spark/interpreter-setting.json ${Z_HOME}/interpreter/spark/interpreter-setting.json
+# COPY --from=zeppelin-distribution /opt/zeppelin/interpreter/jdbc/interpreter-setting.json ${Z_HOME}/interpreter/jdbc/interpreter-setting.json
+# COPY --from=zeppelin-distribution /opt/zeppelin/interpreter/md/interpreter-setting.json ${Z_HOME}/interpreter/md/interpreter-setting.json
```

Comment:
Do you see any side effects when just copy all `/opt/zeppelin/interpreter/*/interpreter-setting.json` files from zeppelin-distribution?

Reply:
I see the following side effect. The Zeppelin server has a long list of interpreters, not all interpreters may work.

Reply:
Does it mean that editing these lines are a mandatory step before build an image using this Dockerfile?

Reply:
It depends, how much images should be maintained by the Zeppelin project. In my opinion Zeppelin should show the way how to extend a image. Zeppelin should only maintain a small set of interpreter.